system/jlib/jsocket.cpp

```diff
@@ -4447,14 +4537,19 @@ class CSocketEpollThread: public CSocketBaseThread
     ~CSocketEpollThread()
     {
         closedummy();
-        ForEachItemIn(i,items) {
-            try {
-                SelectItem &si = items.element(i);
-                epoll_op(epfd, EPOLL_CTL_DEL, si.handle, 0);
-                si.nfy->Release();
-                si.sock->Release();
+        ForEachItemIn(i,items)
+        {
+            try
+            {
+                SelectItem *si = items.element(i);
+                epoll_op(epfd, EPOLL_CTL_DEL, si, 0);
+                si->nfy->Release();
+                si->sock->Release();
+                delete si;
```

Comment:
I agree SelectItem dtor could do nfy/sock Release but I was hesitant to change any of the select method code.